clients/privacy-center/pages/api/fides-js.ts

```diff
@@ -32,6 +35,48 @@ let lastFetched: number = 0;
 // used to disable auto-refreshing if the /custom-asset endpoint is unreachable
 let autoRefresh: boolean = true;
 
+async function retry<T>(
+  func: () => Promise<T> | T,
+  { delay, retries }: { delay: number; retries: number },
+): Promise<T> {
+  try {
+    return await func();
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.warn(
+      `Attempt at getting privacy experience failed, ${retries} remain.`,
+      error,
+    );
+    if (retries > 1) {
+      await new Promise((resolve) => {
+        setTimeout(resolve, delay);
+      });
+      return retry(func, { delay, retries: retries - 1 });
+    }
+    throw error;
+  }
+}
+
+const PREFETCH_RETRY_DELAY = 100;
+const PREFETCH_MAX_RETRIES = 10;
+async function fetchExperienceWithRetry(
+  ...args: Parameters<typeof fetchExperience>
+) {
+  try {
+    return await retry(() => fetchExperience(...args), {
+      delay: PREFETCH_RETRY_DELAY,
+      retries: PREFETCH_MAX_RETRIES,
+    });
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.error(
+      "Privacy Center was not able to prefetch an experience.",
+      error,
+    );
```

Comment:
👌 thanks for the error log, that's preferable.

Just to be clear though, this changes the behaviour to not return an empty experience.

Reply:
I really don't think we ever _want_ to return an empty experience. Originally I was going to return empty after N+1 tries but the more I thought about it the less comfortable I was with doing that.

Reply:
On further review I think we want to return an empty experience ONLY when the experiences endpoint returns an empty array.